Test/Log.Tests.ps1

```diff
@@ -0,0 +1,180 @@
+ 
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-WhiskeyTest.ps1' -Resolve)
+
+$errors = $null
+$warnings = $null
+$infos = $null
+$output = $null
+$failed = $false
+
+function Init
+{
+    $script:errors = $null
+    $script:warnings = $null
+    $script:infos = $null
+    $script:output = $null
+    $Global:Error.Clear()
+}
+
+function ThenFailed
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$ExpectedErrorMessage
+    )
+    
+    $failed | Should -BeTrue
+    $Global:Error | Should -Match $ExpectedErrorMessage
+}
+
+function ThenWroteDebug
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message
+    )
+
+    $output | Should -Match $Message
+}
+
+function ThenWroteError
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message
+    )
+
+    $errors | Should -Match $Message
+}
+
+function ThenWroteInfo
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message
+    )
+
+    $infos | Should -Match $Message
+}
+
+function ThenWroteVerbose
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message
+    )
+
+    $output | Where-Object { $_ -match [regex]::Escape($Message) } | Should -Not -BeNullOrEmpty
+}
+
+function ThenWroteWarning
+{
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message
+    )
+
+    $warnings | Should -Match $Message
+}
+
+function WhenLogging
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [String]$Message,
+
+        [String]$AtLevel
+    )
+
+    $parameter = @{
+        'Message' = $Message
+    }
+
+    if( $AtLevel )
+    {
+        $parameter['Level'] = $AtLevel
+    }
+
+    $script:failed = $false
+    $context = New-WhiskeyTestContext -ForDeveloper -ForBuildRoot $TestDrive.FullName

```

Comment:
Not using `New-WhiskeyTestRoot` ?

Reply:
No. It doesn't use it since the task only writes output.